server/daemon/src/main.rs

```diff
@@ -529,6 +529,11 @@ async fn kanidm_main() -> ExitCode {
         return ExitCode::FAILURE;
     }
 
+    if let Some(threads) = sconfig.thread_count {
```

Comment:
Where are you getting the threads value from? This comparison doesn't make any sense.

Reply:
Should be 
```
std::cmp::min(threads, std::threads::avaliable_parallelism())
```
https://doc.rust-lang.org/stable/std/thread/fn.available_parallelism.html

Reply:
Well that would be the idea since the default for sconfig.thread_count is actually the parallelism, even though if that ever gets changed then I agree it will stop making any sense.
But I also thought who the hell is gonna change it multiple times? So I felt it wasn't *too* bad, but maybe it is

Reply:
But it's not available_parallelism, you're never calling available parallelism. So what is config.threads even set to here? I think you're over-engineering this.